router/_helper_test.gno

```diff
@@ -0,0 +1,450 @@
+	sr.UnstakeToken(tokenId, unwrap)
+}
+
+func wugnotApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	wugnot.Approve(spender, amount)
+}
+
+func gnsApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	gns.Approve(spender, amount)
+}
+
+func barApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	bar.Approve(spender, amount)
+}
+
+func bazApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	baz.Approve(spender, amount)
+}
+
+func fooApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	foo.Approve(spender, amount)
+}
+
+func oblApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	obl.Approve(spender, amount)
+}
+
+func quxApprove(t *testing.T, owner, spender pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(owner)))
+	qux.Approve(spender, amount)
+}
+
+func wugnotTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	wugnot.Transfer(to, amount)
+}
+
+func gnsTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	gns.Transfer(to, amount)
+}
+
+func barTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	bar.Transfer(to, amount)
+}
+
+func bazTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	baz.Transfer(to, amount)
+}
+
+func fooTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	foo.Transfer(to, amount)
+}
+
+func oblTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	obl.Transfer(to, amount)
+}
+
+func quxTransfer(t *testing.T, to pusers.AddressOrName, amount uint64) {
+	t.Helper()
+	std.TestSetRealm(std.NewUserRealm(users.Resolve(admin)))
+	qux.Transfer(to, amount)
+}
+
+// ----------------------------------------------------------------------------
+// ugnot
+
+func ugnotTransfer(t *testing.T, from, to std.Address, amount uint64) {
+	t.Helper()
+
+	std.TestSetRealm(std.NewUserRealm(from))
+	std.TestSetOrigSend(std.Coins{{ugnotDenom, int64(amount)}}, nil)
+	banker := std.GetBanker(std.BankerTypeRealmSend)
```

Comment:
gnolang/gno merged new pr relates to banker / coin issue
- https://github.com/gnolang/gno/pull/3239

need to check